### PR TITLE
feat: add listings and leads analytics api

### PR DESCRIPTION
**Pull request by ufkaya**
[CAR-5242 - Cockpit: listings and lead number](https://autoricardo.atlassian.net/browse/CAR-5242)

### CODE REVIEW

File: src/services/analytics.ts [hunk 1]
```diff
@@ -14,3 +18,32 @@ export const fetchAnalyticsData = async (
...
+    query,
+  })
+}
+
+export const fetchListingsAnalytics = async ({
+  dealerId,
+  dimensions,
+}: {
+  dealerId: number
+  dimensions?: string[]
```
Comment (masone) [inline src/services/analytics.ts:43]: is it worth typing the dimensions here? or is it an undefined set of strings?

File: src/services/analytics.ts [hunk 2]
```diff
@@ -14,3 +18,32 @@ export const fetchAnalyticsData = async (
...
 }
+
+export const fetchLeadsAnalytics = async ({
+  dealerId,
+  dimensions,
+  query,
+}: {
+  dealerId: number
+  dimensions?: string[]
+  query?: { verificationDateFrom: string }
```
Comment (masone) [inline src/services/analytics.ts:29]: is this the only query parameter available on this call?

**Review by masone — APPROVED** [review]

File: src/services/analytics.ts (hunk 1 above)
Comment (ufkaya) [reply src/services/analytics.ts:43]: I need it in case we want to pass the dimension like `conditionType`

**Comment by masone** [comment]
@Averethel @lkappeler you'll need to rebase on these :)

**Comment by lkappeler** [comment]
Jup did that

**Pull request merged**